### PR TITLE
module: fix circular reference to mod

### PR DESCRIPTION
**Pull request by yorkie**
The module 1 like this:

```js
require('./module-2');
```

and module 2 is:

```js
require('./module-1');
```

Just write cache before compiling to fix issue.

### CODE REVIEW

**Comment by LaszloLango** [comment]
Please add a test for this.

**Comment by yorkie** [comment]
@LaszloLango Thanks, tests is added now :)

**Comment by glistening** [comment]
@yorkie Could you please fix travis error?

> Detected style problems:
> /home/travis/build/Samsung/iotjs/src/js/module.js:190: trailing whitespace

**Comment by yorkie** [comment]
Sorry about missing this message, will fix tonight :)

**Comment by zherczeg** [comment]
Please resolve the conflicts.

**Comment by yorkie** [comment]
Resolved :)

**Comment by yichoi** [comment]
@yorkie rebasing instead of merging is needed.

File: src/js/module.js [hunk 1]
```diff
@@ -206,6 +206,8 @@ Module.load = function(id, parent) {
 
   module.filename = modPath;
   module.dirs = [modPath.substring(0, modPath.lastIndexOf('/') + 1)];
+  iotjs_module_t.cache[modPath] = module;
```
Comment (LaszloLango) [inline src/js/module.js]: Module.cache[modPath] = module;

**Comment by hs0225** [comment]
I close this PR because there is no action for a long time. And I'll resolve comments and make another PR.

**Pull request closed without merging**